### PR TITLE
Add branching and releasing model doc

### LINKED ISSUE #879: DOC - Add documentation for the branching and releasing model
The "Contribute" section would benefit from having documentation around the branching model of the project and how to cut releases.

┆Issue is synchronized with this [Jira Bug](https://k8ssandra.atlassian.net/browse/K8SSAND-569) by [Unito](https://www.unito.io)

### PR DESCRIPTION
**Pull request by adejanovski**
**What this PR does**:
Adds documentation explaining the branching model of the project and detailing how to perform releases.

**Which issue(s) this PR fixes**:
Fixes #879

**Checklist**
- [x] Changes manually tested
- [ ] Automated Tests added/updated
- [x] Documentation added/updated
- [ ] CHANGELOG.md updated (not required for documentation PRs)
- [x] CLA Signed:  [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

File: docs/content/en/contribute/branching_model.md [hunk 1]
```diff
@@ -0,0 +1,359 @@
...
+## main
+
+The main branch contains the next release and all PRs adding new features or fixes for the upcoming major/minor release should be merged in it.
+
+**This branch is protected against force push to prevent rewriting its history.**
+
+Direct pushes to main should be avoided as much as possible to allow peer review through the PR process. This also allows unit/integration tests to run before a commit is pushed to **main** as we cannot rewrite history.
+
+
+## &lt;feature>
```
Comment (bradfordcp) [inline docs/content/en/contribute/branching_model.md:41]: Should this use `&rt;`? For the closing brace?

```suggestion
## &lt;feature&rt;
```

**Comment by bradfordcp** [comment]
Consider moving this file to `docs/content/en/contribute/branching_model/_index.md` with all the images under `docs/content/en/contribute/branching_model/`.

**Comment by bradfordcp** [comment]
Hey @adejanovski I've pushed a commit up with a bunch of slight formatting tweaks. Let me know what you think.

File: docs/content/en/contribute/branching_model/_index.md [hunk 2]
```diff
@@ -0,0 +1,333 @@
...
+
+The main branch contains the next release and all PRs adding new features or fixes for the upcoming major/minor release should be merged in it.
+
+**This branch is protected against force push to prevent rewriting its history.**
+
+Direct pushes to main should be avoided as much as possible to allow peer review through the PR process. This also allows unit/integration tests to run before a commit is pushed to **main** as we cannot rewrite history.
+
+### &lt;feature&gt;
+
+Feature branches are branched off the **main** branch. Feature branches should be created **on our personal forks**.
```
Comment (bradfordcp) [inline docs/content/en/contribute/branching_model/_index.md:34]: Do we want to link to GH docs around forking projects?

File: docs/content/en/contribute/branching_model.md (hunk 1 above)
Comment (bradfordcp) [reply docs/content/en/contribute/branching_model.md]: For the record &rt; isn't a thing... &gt; is. I've made this change in the most recent commit.

File: docs/content/en/contribute/branching_model/_index.md (hunk 2 above)
Comment (bradfordcp) [reply docs/content/en/contribute/branching_model/_index.md:34]: https://docs.github.com/en/github/getting-started-with-github/quickstart/fork-a-repo maybe

Comment (adejanovski) [reply docs/content/en/contribute/branching_model/_index.md:34]: Sure thing, I'll add this.

**Comment by adejanovski** [comment]
> Hey @adejanovski I've pushed a commit up with a bunch of slight formatting tweaks. Let me know what you think.

LGTM, thanks for fixing the doc 🙏 
I've pushed a commit to add the link to GH docs on forking projects.

**Comment by adejanovski** [comment]
@bradfordcp, looks good to you now?

**Review by jdonenine — APPROVED** [review]

**Pull request merged**